cwltool/provenance_profile.py

```diff
@@ -296,6 +297,24 @@ def record_process_end(
         self.generate_output_prov(outputs, process_run_id, process_name)
         self.document.wasEndedBy(process_run_id, None, self.workflow_run_uri, when)
 
+    def _add_nested_annotations(self, annotation_key, annotation_value, e: ProvEntity) -> ProvEntity:
+        """Propagate input data annotations to provenance."""
+        # Change https:// into http:// first
+        schema2_uri = "https://schema.org/" 
+        if schema2_uri in annotation_key:
+            annotation_key = SCHEMA[annotation_key.replace(schema2_uri, '')].uri 
+        
+        if not isinstance(annotation_value, (MutableSequence, MutableMapping)):            
+            e.add_attributes({annotation_key: str(annotation_value)})
+        else:
+            nested_id = uuid.uuid4().urn
+            nested_entity = self.document.entity(nested_id)
+            e.add_attributes({annotation_key: nested_entity.identifier})
+            for nested_key in annotation_value:
+                nested_value = annotation_value[nested_key] 
```

Comment:
This gives an error when annotation_value is a CommentedSeq 
= Input file has array of annotations like this:
```
s:hasPart:
- s:identifier: <identifier1>
- s:identifier: <identifier2>
```

Reply:
Add if CommentedSeq